dialoguekit/participant/participant.py

```diff
@@ -54,12 +54,16 @@ def connect_dialogue_manager(
         """
         self._dialogue_manager = dialogue_manager
 
+    @abstractmethod
     def receive_utterance(
         self, annotated_utterance: AnnotatedUtterance
```

Comment:
Should we not assume simply an Utterance here? For certain participants it might be an AnnotatedUtterance, but that is enough to restrict in the derived class.

Reply:
Yes, I changed `AnnotatedUtterance` by `Utterance`. The derived class cannot use the type `AnnotatedUtterance ` without violating the Liskov substitution principle (see explanation [here](https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides)).

Reply:
There are workarounds -- we can cross that bridge when we get there. But I'm a bit confused now: we are not using mypy for type checking, are we?

Reply:
We are.

Reply:
Where does that come from? It is not part of our [default setup](https://github.com/iai-group/guidelines/tree/main/python) nor [project template](https://github.com/iai-group/template-project). I missed when it was added and I'd argue that we don't want it unless there are specific reasons for having it.

Reply:
Resolved internally.